### ISSUE TITLE
Remove Icon not permanent

### ISSUE THREAD

**Issue by RealRaven2000** [RealRaven2000/QuickFolders#449]
After removing a custom icon from folder tree using the context menu - the icon returns again after restart

![image](https://github.com/RealRaven2000/QuickFolders/assets/6314674/be8a7473-d8be-4f07-9905-79e7a80e4c6a)

I couldn't reproduce this issue on my test accounts so far.

**Comment by blechma** [comment]
Thank you very much!

**Comment by RealRaven2000** [comment]
> Thank you very much!

Thanks for sending the test data (off github) - I will investigate over the week.

**Comment by blechma** [comment]
😊

**Comment by RealRaven2000** [comment]
Here is the portable installation (account name redacted)

![image](https://github.com/RealRaven2000/QuickFolders/assets/6314674/51673b01-4b6a-4f16-b5a7-0d4a47520161)

Is the test folder the one with the custom icon? I guess I cannot see it because it references an icon on your hard drive....

**Comment by RealRaven2000** [comment]
I tried to restore the Test folder but the (invisible) custom icon returns after restart, hopefully I can debug the problem.

**Comment by RealRaven2000** [comment]
While debugging, this is the code during removal of the icon:

![image](https://github.com/RealRaven2000/QuickFolders/assets/6314674/ee4e08aa-6665-41c2-ac9e-849394e78c1c)

values before:

```
folder.getStringProperty("folderIcon")
"folderIcon_server1_******_******_test_imap_web_de_Test"
folder.getStringProperty("iconURL")
"url(file:///C:/Users/Public/Pictures/Icons/Action.ico)" 

```

values after execution:

```
folder.getStringProperty("folderIcon")
"noIcon"
folder.getStringProperty("iconURL")
"" 
```

so it looks like these properties are set in the session, as expected but for some reason they are not stored to the actual folder data by Thunderbird. I need to investigate whether there is a way to force submitting them or whether it needs an actual change in the contents (e.g. adding or removing an email) to trigger this behavior. As far as I know is no explicite, scriptable "store()" or "persist()" function to make folder state permanent. It could also be a special behavior of Thunderbird portable?

**Comment by RealRaven2000** [comment]
I made some more tests and found out it behaved the same with my own icons, but only on the test folder. I was able to remove the icon from the Inbox folder permanently. So there must be a fundamental difference between both (probably above and beyond that folder flags). I will try with some subfolder under test next. I will also try  what happens if I move Test to be a child folder of inbox.

**Comment by RealRaven2000** [comment]
I couldn't create subfolders within inbox (maybe this is because I have no access to the original IMAP server) but I was able to generate a new Test 2 folder on the same level as Test (root). With that I was able to add and remove icons permanently.

**Comment by RealRaven2000** [comment]
Unfortunately  I cannot move folders with in the tree (may also be a problem of Imap access). Still it is weird to me that I am able to modify my own folder and the inbox and persist this but not with the original Test folder.... does it exist on the original server?

**Comment by RealRaven2000** [comment]
CAn you test again what happens if you create a new folder alongside one of the "misbehaving" folders, does it also allow removal of the icon or does it behave the same way as the "Test" folder you sent in the portable profile?

**Comment by blechma** [comment]
> Unfortunately I cannot move folders with in the tree (may also be a problem of Imap access). Still it is weird to me that I am able to modify my own folder and the inbox and persist this but not with the original Test folder.... does it exist on the original server?

Yes, the "Test" and the "Test 2" folder exist on the original server. I will send you the datas to access to the original folder per mail.

**Comment by RealRaven2000** [comment]
> Yes, the "Test" and the "Test 2" folder exist on the original server. I will send you the datas to access to the original folder per mail.

no thanks - no need to do that right now, I think I can manage with the profile as is. I would rather not have access to your data - I can find out with the colleagues from Thunderbird on the development chat on Thursday

**Comment by blechma** [comment]
> CAn you test again what happens if you create a new folder alongside one of the "misbehaving" folders, does it also allow removal of the icon or does it behave the same way as the "Test" folder you sent in the portable profile?

The new folder "Test3" works normal. It is not a "misbehaving" folder

**Comment by blechma** [comment]
> > Yes, the "Test" and the "Test 2" folder exist on the original server. I will send you the datas to access to the original folder per mail.
> 
> no thanks - no need to do that right now, I think I can manage with the profile as is. I would rather not have access to your data - I can find out with the colleagues from Thunderbird on the development chat on Thursday

I have send the datas before I have read your statement. But it is no problem. It is a "Test"-account
